### PR TITLE
Login Test Suite Espresso

### PR DESCRIPTION
**Pull request by jesmrec**
UI testing suite on the ´AuthenticatorActivity´ activity to check the login feature.

Included:
- Positive and negative test cases. 
- Checks in both orientations 
- Accounts Manager: to add/remove ownCloud accounts from device.

To do:
- SAML Servers

### CODE REVIEW

File: androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java [hunk 1]
```diff
@@ -19,59 +20,88 @@
...
+    private Context targetContext = null;
 
+    private String testUser = null;
+    private String testUser2 = null;
+    private String testPassword = null;
+    private String testPassword2 = null;
+    private String testServerURL = null;
+    private int trusted = -1;
+
+    /*
```
Comment (davivel) [inline androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java]: This is begging for an enum ;)

File: androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java [hunk 2]
```diff
@@ -81,59 +111,460 @@ protected Intent getActivityIntent() {
...
+
+            // Check that login button keeps on being disabled
+            onView(withId(R.id.buttonOK))
+                    .check(matches(not(isEnabled())));
+
+            // Check that SSL server is not trusted
+            onView(withId(R.id.server_status_text))
+                    .check(matches(withText(R.string.ssl_certificate_not_trusted)));
+
+            Log_OC.i(LOG_TAG, "Test not accept not secure passed");
```
Comment (davivel) [inline androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java]: No assert in this test?

Comment (jesmrec) [inline androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java]: yes, line 179 contains the `matches` assertion. This is a negative test case, so only is checked the error message.

File: androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java (hunk 1 above)
Comment (jesmrec) [inline androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java]: you mean, set the `trusted` as an enum with the values in the comment.

File: androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java (hunk 2 above)
Comment (davivel) [inline androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java]: :+1:

File: androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java (hunk 1 above)
Comment (davivel) [inline androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java]: Yes, defining an enum type with those values an use the constants of the enum to make it easier to read all over the code.

Example:https://github.com/owncloud/android/blob/master/src/com/owncloud/android/datamodel/UploadsStorageManager.java#L49

Comment (jesmrec) [inline androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java]: @davivel done!! please review

File: androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java [hunk 3]
```diff
@@ -543,20 +581,20 @@ private void setFields (String connectionString, String username, String passwor
      */
     private void checkStatusMessage(){
 
-        switch (trusted){
-            case (-1):
+        switch (servertype.getStatus()){
```
Comment (davivel) [inline androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java]: An enum can be used directly in a ```switch``` statement. You don't need to get its internal value, just write

```switch (servertype)```

and then the cases as

```case HTTP:```,  etc...

File: androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java [hunk 4]
```diff
@@ -81,59 +138,471 @@ protected Intent getActivityIntent() {
...
+    public void init() {
+        Bundle arguments = InstrumentationRegistry.getArguments();
+
+        testUser = arguments.getString("TEST_USER");
+        testUser2 = arguments.getString("TEST_USER2");
+        testPassword = arguments.getString("TEST_PASSWORD");
+        testPassword2 = arguments.getString("TEST_PASSWORD2");
+        testServerURL = arguments.getString("TEST_SERVER_URL");
+        trusted = Integer.parseInt(arguments.getString("TRUSTED"));
+        switch (trusted) {
```
Comment (davivel) [inline androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java]: I would move this logic to get an enum from an int to a static method in the enum, similar to the method ```fromValue``` here: https://github.com/owncloud/android/blob/master/src/com/owncloud/android/datamodel/UploadsStorageManager.java#L76

That way you could read the servertype here, or at any other place, directly as

```servertype = ServerType.fromValue(Integer.parseInt(arguments.getString("TRUSTED")));```

and the field ```trusted``` wouldn't be needed anymore.

File: androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java (hunk 3 above)
Comment (jesmrec) [inline androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java]: done!

File: androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java (hunk 4 above)
Comment (jesmrec) [inline androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java]: done, thanks!

**Comment by davivel** [comment]
In.!

**Pull request merged**